### PR TITLE
add new package mqueue version 0.5.1

### PR DESCRIPTION
**Pull request by mwweissmann**

### CODE REVIEW

**Comment by dsheets** [comment]
This needs a dependency on `base-bytes` to support OCaml < 4.02 and `available: [ os != "darwin" ]` to signal the unavailability of mqueues on OS X.

**Comment by mwweissmann** [comment]
ok, I give up: where is the type definition for "bytes" in pre-4.02?

File: packages/mqueue/mqueue.0.5.2-0/opam [hunk 1]
```diff
@@ -0,0 +1,21 @@
+opam-version: "1.2"
+maintainer: "Markus W. Weissmann <markus.weissmann@in.tum.de>"
+authors: "Markus W. Weissmann <markus.weissmann@in.tum.de>"
+homepage: "http://mqueue.forge.ocamlcore.org/"
+bug-reports: "http://mqueue.forge.ocamlcore.org/"
+dev-repo: "git://mqueue.forge.ocamlcore.org/"
```
Comment (Chris00) [inline packages/mqueue/mqueue.0.5.2-0/opam]: I do not think this is a valid repository location.

**Comment by mwweissmann** [comment]
The repo is currently on a private server, but opam denies not providing a "dev-repo" field.

File: packages/mqueue/mqueue.0.5.2-0/opam [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+license: "MIT"
+build: [
+  ["ocaml" "setup.ml" "-configure"]
```
Comment (Chris00) [inline packages/mqueue/mqueue.0.5.2-0/opam]: I think you need to add ` "--prefix" prefix`.  [oasis2opam](https://github.com/ocaml/oasis2opam) can automate this.

**Comment by Chris00** [comment]
You may want to replace your `_oasis` file with
```
OASISFormat: 0.4
Name:        mqueue
Version:     0.5.2
Synopsis:    POSIX message queues
Description:  The mqueue library is a small OCaml library providing types
  and functions for using POSIX message queues. The types and functions
  try to stay close to the C API. This library has been tested on Linux,
  but QNX, Solaris etc. should work, too. Mac OS X on the other hand does
  NOT provide POSIX message queues (OS X 10.10). 
Authors:     Markus Weissmann
Maintainers: Markus W. Weissmann <markus.weissmann@in.tum.de>
Homepage:    http://mqueue.forge.ocamlcore.org/
License:     MIT
Plugins:     META (0.4)

Library "mqueue"
  Path:       src
  CSources:   mqueue_stubs.c
  CCLib:      -lrt
  CCOpt:      -O2 -Wall -Wextra -Wno-unused-parameter
  BuildTools: ocamlbuild
  Modules:    Mqueue
  BuildDepends: unix, bytes

Document doclib
  Title:                API reference for POSIX message queues
  Type:                 ocamlbuild (0.3)
  Install:              false
  BuildTools+:          ocamlbuild, ocamldoc
  XOCamlbuildPath:      src
  XOCamlbuildLibraries: mqueue

Executable "mqtest"
  Path:       test
  MainIs:     test.ml
  Build$:     flag(tests)
  CompiledObject: best
  BuildTools: ocamlbuild
  Install: false
  BuildDepends: mqueue

Test "mqtest"
  Run$: flag(tests)
  TestTools: mqtest
  Command: $mqtest
  WorkingDirectory: test
```
Some additional metadata is provided.

**Comment by Chris00** [comment]
You have to replace `bytes` by `Bytes.t` (only a `Bytes` module is provided by `ocamlfind`).

**Pull request merged**